workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.workflows;
+
+// [START workflows_api_quickstart]
+
+// Imports the Google Cloud client library
+
+import com.google.cloud.workflows.executions.v1.CreateExecutionRequest;
+import com.google.cloud.workflows.executions.v1.Execution;
+import com.google.cloud.workflows.executions.v1.ExecutionsClient;
+import com.google.cloud.workflows.executions.v1.WorkflowName;
+
+import java.io.IOException;
+
+/**
+ * Cloud Workflows API sample. Executes a workflow and waits for results.
+ * Example usage:
+ * GOOGLE_CLOUD_PROJECT=myProject \
+ * mvn package exec:java -Dexec.mainClass='com.example.workflows.WorkflowsQuickstart'
+ */
+public class WorkflowsQuickstart {
+  private static final String GOOGLE_CLOUD_PROJECT = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static String LOCATION = System.getenv("LOCATION");
+  private static String WORKFLOW = System.getenv("WORKFLOW");
+
+  public static String workflowsQuickstart(String projectId, String location, String workflow) throws InterruptedException, IOException {
+    // Execute workflow
+    try (ExecutionsClient workflowExecutionsClient = ExecutionsClient.create()) {
+      WorkflowName parent = WorkflowName.of(projectId, location, workflow);
+      Execution initialExecution = Execution.newBuilder().build();
+
+      // Creates the execution object.
+      Execution createExecutionRes = workflowExecutionsClient
+          .createExecution(parent, initialExecution);
+
+      String executionName = createExecutionRes.getName();
+      System.out.printf("Created execution: %s%n", executionName);
+
+      // Wait for execution to finish, then print results.
+      boolean executionFinished = false;
+      long backoffDelay = 1_000; // Start wait with delay of 1,000 ms
+      final long BACKOFF_TIMEOUT = 10 * 60_000; // Time out at 10 minutes
+      System.out.println("Poll for results...");
+      while (!executionFinished) {
+        Execution execution = workflowExecutionsClient.getExecution(executionName);
+        executionFinished = execution.getState() != Execution.State.ACTIVE;
+
+        // We've passed the max
+        if (backoffDelay > BACKOFF_TIMEOUT) {
+          return "";
```

Comment:
Perhaps throw an error here instead? @lesv would you agree?

Reply:
Yes - in general, Idiomatic Java shouldn't be returning errors, unless they are common.  If it's an exceptional situation, as this one looks to be, then we throw an exception.  Effective Java 3e Ch 69/70